projects/media-viewer/src/lib/viewers/image-viewer/image-viewer.component.ts

```diff
@@ -37,12 +38,28 @@ export class ImageViewerComponent implements OnChanges {
 
   drawStartX = -1;
   drawStartY = -1;
+  // local array of any subscriptions so that we can tidy them up later
+  private subscriptions: Subscription[] = [];
 
   constructor(
     private readonly printService: PrintService,
-    private readonly api: AnnotationApiService
+    private readonly api: AnnotationApiService,
+    private readonly toolbarEventsService: ToolbarEventsService
   ) { }
 
+  ngOnInit(): void {
+    // Listen for any changes invoked on the toolbar events Service and initialise any default behaviour state
+    this.subscriptions.push(this.toolbarEventsService.drawMode.subscribe((toggleValue) => {
```

Comment:
oh look... it's that boilerplate again :D it's feeling a bit wet in here.

Reply:
lol...

Reply:
in this scenario… i believe this is the right pattern. Its responsible for updating an internal property within the component. 
I can’t see this being an async pipe scenario. @linusnorton this is the pattern i'd like for us to use in scenarios where state of the component is being changed